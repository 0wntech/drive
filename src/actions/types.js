export const LOGIN = 'login';
export const LOGIN_SUCCESS = 'login_success';
export const LOGIN_FAIL = 'login_fail';

export const FETCH_USER = 'fetch_user';
export const FETCH_USER_SUCCESS = 'fetch_user_success';
export const FETCH_USER_FAIL = 'fetch_user_fail';

export const SET_WEBID = 'set_webid';
export const SET_CURRENT_PATH = 'set_current_path';

export const LOAD_CURRENT_ITEMS = 'load_current_items';
export const SET_CURRENT_ITEMS = 'set_current_items';

export const FETCH_FRIENDS = 'fetch_friends';
export const FETCH_FRIENDS_SUCCESS = 'fetch_friends_success';
export const FETCH_FRIENDS_FAIL = 'fetch_friends_fail';

export const FETCH_CURRENT_ITEM = 'fetch_current_item';
export const FETCH_CURRENT_ITEM_SUCCESS = 'fetch_current_item_success';
export const FETCH_CURRENT_ITEM_FAIL = 'fetch_current_item__fail';

export const GET_SESSION = 'get_session';
export const GET_SESSION_SUCCESS = 'get_session_success';
export const GET_SESSION_FAIL = 'get_session_fail';

export const FETCH_NOTIFICATIONS = 'fetch_notifications';
export const FETCH_NOTIFICATIONS_SUCCESS = 'fetch_notifications_success';
export const FETCH_NOTIFICATIONS_FAILURE = 'fetch_notifications_failure';

export const SET_SELECTION = 'set_selection';

export const SEND_NOTIFICATION = 'send_notification';
export const SEND_NOTIFICATION_SUCCESS = 'send_notification_success';
export const SEND_NOTIFICATION_FAILURE = 'send_notification_failure';

export const FETCH_IDPS = 'fetch_idps';
export const FETCH_IDPS_SUCCESS = 'fetch_idps_success';
export const FETCH_IDPS_FAILED = 'fetch_idps_failure';

export const DELETE_ITEMS = 'delete_items';
export const DELETE_ITEMS_SUCCESS = 'delete_items_success';
export const DELETE_ITEMS_FAILURE = 'delete_items_failure';

export const COPY_ITEMS = 'copy_items';
export const PASTE_ITEMS = 'paste_items';
export const PASTE_ITEMS_SUCCESS = 'paste_items_success';
export const PASTE_ITEMS_FAILURE = 'paste_items_failure';

export const FETCH_CONTACTS = 'fetch_contacts';
export const FETCH_CONTACTS_SUCCESS = 'fetch_contacts_success';
export const FETCH_CONTACTS_FAILURE = 'fetch_contacts_failure';

export const ADD_CONTACT = 'add_contact';
export const ADD_CONTACT_SUCCESS = 'add_contact_success';
export const ADD_CONTACT_FAILURE = 'add_contact_failure';

export const REMOVE_CONTACT = 'remove_contact';
export const REMOVE_CONTACT_SUCCESS = 'remove_contact_success';
export const REMOVE_CONTACT_FAILURE = 'remove_contact_failure';

export const UPDATE_PROFILE = 'delete_items';
export const UPDATE_PROFILE_SUCCESS = 'update_profile_success';
export const UPDATE_PROFILE_FAILURE = 'update_profile_failure';

export const CHANGE_PROFILE_PHOTO = 'change_profile_photo';
export const CHANGE_PROFILE_PHOTO_SUCCESS = 'change_profile_photo_success';
export const CHANGE_PROFILE_PHOTO_FAILURE = 'change_profile_photo_failure';

export const SET_CURRENT_CONTACT = 'set_current_contact';

export const RENAME_ITEM = 'rename_item';
export const RENAME_ITEM_SUCCESS = 'rename_item_success';
export const RENAME_ITEM_FAILURE = 'rename_item_failure';

export const SEARCH_CONTACT = 'search_contact';
export const SEARCH_CONTACT_SUCCESS = 'search_contact_success';
export const SEARCH_CONTACT_FAILURE = 'search_contact_failure';
