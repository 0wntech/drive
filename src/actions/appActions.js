import {
    FETCH_CURRENT_ITEM,
    FETCH_CURRENT_ITEM_SUCCESS,
    FETCH_CURRENT_ITEM_FAIL,
    SET_CURRENT_PATH,
    SET_SELECTION,
    FETCH_NOTIFICATIONS,
    FETCH_NOTIFICATIONS_SUCCESS,
    FETCH_NOTIFICATIONS_FAILURE,
    SEND_NOTIFICATION,
    SEND_NOTIFICATION_SUCCESS,
    SEND_NOTIFICATION_FAILURE,
    FETCH_IDPS,
    FETCH_IDPS_SUCCESS,
    FETCH_IDPS_FAILED,
    OPEN_CONSENT_WINDOW,
    CLOSE_CONSENT_WINDOW,
    DELETE_ITEMS,
    DELETE_ITEMS_SUCCESS,
    DELETE_ITEMS_FAILURE,
    COPY_ITEMS,
    PASTE_ITEMS,
    PASTE_ITEMS_SUCCESS,
    PASTE_ITEMS_FAILURE,
    OPEN_RENAME_WINDOW,
    CLOSE_RENAME_WINDOW,
    RENAME_ITEM,
    RENAME_ITEM_SUCCESS,
    RENAME_ITEM_FAILURE,
    OPEN_CREATE_FILE_WINDOW,
    CLOSE_CREATE_FILE_WINDOW,
    CREATE_FILE,
    CREATE_FILE_FAILURE,
    CREATE_FILE_SUCCESS,
    UPDATE_FILE,
    UPDATE_FILE_SUCCESS,
    UPDATE_FILE_FAILURE,
    DOWNLOAD_FILE,
    DOWNLOAD_FILE_SUCCESS,
    DOWNLOAD_FILE_FAILURE,
    OPEN_CREATE_FOLDER_WINDOW,
    CLOSE_CREATE_FOLDER_WINDOW,
    CREATE_FOLDER,
    CREATE_FOLDER_SUCCESS,
    CREATE_FOLDER_FAILURE,
} from './types';
import auth from 'solid-auth-client';
import fileUtils from '../utils/fileUtils';
import PodClient from 'ownfiles';
import mime from 'mime';
import url from 'url';
import FileSaver from 'file-saver';
import { convertFolderUrlToName, convertFileUrlToName } from '../utils/url';

export const setCurrentPath = (newPath, options = {}) => {
    return (dispatch) => {
        dispatch({ type: SET_CURRENT_PATH, payload: newPath });
        dispatch({ type: SET_SELECTION, payload: [] });
        if (options.noFetch) {
            return dispatch({
                type: FETCH_CURRENT_ITEM_SUCCESS,
                payload: { body: '', url: newPath },
            });
        } else {
            return dispatch(fetchCurrentItem(newPath, newPath.endsWith('/')));
        }
    };
};

export const fetchCurrentItem = (itemUrl, folder = false) => {
    return (dispatch) => {
        dispatch({ type: FETCH_CURRENT_ITEM });
        const fileClient = new PodClient({
            podUrl: 'https://' + url.parse(itemUrl).host + '/',
        });
        const options = {};
        if (folder) {
            options.auth = auth;
            options.headers = { Accept: 'text/turtle' };
        }
        return fileClient
            .read(itemUrl, options)
            .then((item) => {
                if (item && item.folders) {
                    const fileNames = item.files.map((file) => {
                        return convertFileUrlToName(file);
                    });
                    const folderNames = item.folders.map((folder) => {
                        return convertFolderUrlToName(folder);
                    });
                    dispatch({
                        type: FETCH_CURRENT_ITEM_SUCCESS,
                        payload: {
                            files: fileNames,
                            folders: folderNames,
                        },
                    });
                } else if (item && typeof item === 'string') {
                    dispatch({
                        type: FETCH_CURRENT_ITEM_SUCCESS,
                        payload: { body: item, url: itemUrl },
                    });
                }
            })
            .catch((error) =>
                dispatch({
                    type: FETCH_CURRENT_ITEM_FAIL,
                    payload: error,
                })
            );
    };
};

export const downloadFile = (file) => {
    return (dispatch) => {
        dispatch({ type: DOWNLOAD_FILE });
        const fileClient = new PodClient({});
        fileClient
            .read(file)
            .then((result) => {
                const fileType = mime.getType(file);
                if (fileType.includes('image')) {
                    console.log('saving as image');
                    FileSaver.saveAs(file, convertFileUrlToName(file));
                    dispatch({ type: DOWNLOAD_FILE_SUCCESS });
                } else {
                    const blob = new Blob([result], {
                        type: mime.getType(file),
                    });
                    FileSaver.saveAs(blob, convertFileUrlToName(file));
                    dispatch({ type: DOWNLOAD_FILE_SUCCESS });
                }
            })
            .catch((err) => {
                dispatch({ type: DOWNLOAD_FILE_FAILURE, payload: err });
            });
    };
};

export const fetchNotifications = (webId) => {
    return (dispatch) => {
        dispatch({ type: FETCH_NOTIFICATIONS });
        fileUtils
            .getNotificationFiles(webId)
            .then((notifications) => {
                dispatch({
                    type: FETCH_NOTIFICATIONS_SUCCESS,
                    payload: notifications,
                });
            })
            .catch((err) => {
                dispatch({
                    type: FETCH_NOTIFICATIONS_FAILURE,
                });
            });
    };
};

export const updateFile = (file, body) => {
    return (dispatch) => {
        dispatch({ type: UPDATE_FILE });
        const fileClient = new PodClient({
            podUrl: 'https://' + url.parse(file).host + '/',
        });
        if (body !== '') {
            fileClient
                .update(file, body)
                .then(() => {
                    dispatch({
                        type: UPDATE_FILE_SUCCESS,
                        payload: { url: file, body: body },
                    });
                })
                .catch((err) => {
                    dispatch({
                        type: UPDATE_FILE_FAILURE,
                        payload: err,
                    });
                });
        } else {
            dispatch({
                type: UPDATE_FILE_FAILURE,
                payload: "File can't be set to empty",
            });
        }
    };
};

export const sendNotification = (to, notification) => {
    return (dispatch) => {
        dispatch({ type: SEND_NOTIFICATION });
        fileUtils
            .sendNotification(to, notification)
            .then(() => {
                dispatch({
                    type: SEND_NOTIFICATION_SUCCESS,
                });
            })
            .catch((err) => {
                dispatch({
                    type: SEND_NOTIFICATION_FAILURE,
                    payload: err,
                });
            });
    };
};

export const setSelection = (selection) => {
    return { type: SET_SELECTION, payload: selection };
};

export const fetchIdps = () => {
    return (dispatch) => {
        dispatch({ type: FETCH_IDPS });
        const request = { method: 'GET' };
        fetch('https://solid.github.io/solid-idp-list/services.json', request)
            .then((response) => {
                response.json().then((idps) => {
                    dispatch({ type: FETCH_IDPS_SUCCESS, payload: idps.idps });
                });
            })
            .catch((err) => {
                dispatch({ type: FETCH_IDPS_FAILED, payload: err });
            });
    };
};

export const deleteItems = (items, currentPath = '/') => {
    return (dispatch) => {
        dispatch({ type: DELETE_ITEMS });
        fileUtils
            .deleteItems(items)
            .then(() => {
                // To avoid reloading when not everything has been deleted
                setTimeout(() => {
                    dispatch({ type: DELETE_ITEMS_SUCCESS });
                    dispatch(setCurrentPath(currentPath));
                }, 2000);
            })
            .catch((err) => {
                dispatch({ type: DELETE_ITEMS_FAILURE, payload: err });
            });
    };
};

export const copyItems = (items) => {
    return (dispatch) => {
        dispatch({ type: COPY_ITEMS, payload: items });
    };
};

export const pasteItems = (items, location) => {
    return (dispatch) => {
        dispatch({ type: PASTE_ITEMS });
        auth.currentSession()
            .then((session) => {
                const fileClient = new PodClient({ podUrl: session.webId });
                const paste = new Promise((resolve, reject) => {
                    items.map((item, index) => {
                        if (index === items.length - 1) {
                            return fileClient.copy(item, location).then(() => {
                                resolve();
                            });
                        } else {
                            return fileClient.copy(item, location);
                        }
                    });
                    Promise.all(items).catch((err) => {
                        reject(err);
                    });
                });
                Promise.resolve(paste)
                    .then(() => {
                        dispatch({ type: PASTE_ITEMS_SUCCESS });
                        dispatch(setCurrentPath(location));
                    })
                    .catch((err) => {
                        dispatch({ type: PASTE_ITEMS_FAILURE, payload: err });
                    });
            })
            .catch((err) => {
                dispatch({ type: PASTE_ITEMS_FAILURE, payload: err });
            });
    };
};

export const createFile = function(name, path) {
    return (dispatch) => {
        dispatch({ type: CREATE_FILE });
        const contentType = mime.getType(name);
        const fileClient = new PodClient({
            podUrl: 'https://' + url.parse(name).host + '/',
        });
        return fileClient
            .create(path + name, { contentType: contentType })
            .then(() => {
                dispatch({ type: CREATE_FILE_SUCCESS });
                dispatch(setCurrentPath(path));
            })
            .catch((err) => {
                dispatch({ type: CREATE_FILE_FAILURE, payload: err });
            });
    };
};

export const renameItem = function(renamedItem, value) {
    return (dispatch) => {
        dispatch({ type: RENAME_ITEM });
        auth.currentSession().then((session) => {
            const fileClient = new PodClient({ podUrl: session.webId });
            const rename = new Promise((resolve, reject) => {
                if (renamedItem.endsWith('/')) {
                    fileClient
                        .renameFolder(renamedItem, value)
                        .then(() => {
                            resolve();
                        })
                        .catch((err) => {
                            reject(err);
                        });
                } else {
                    fileClient
                        .renameFile(renamedItem, value)
                        .then(() => {
                            resolve();
                        })
                        .catch((err) => {
                            reject(err);
                        });
                }
            });
            Promise.resolve(rename)
                .then(() => {
                    let location = renamedItem.split('/');
                    if (renamedItem.endsWith('/')) {
                        location.pop();
                        location.pop();
                    } else {
                        location.pop();
                    }

                    location = location.join('/');
                    dispatch({ type: RENAME_ITEM_SUCCESS });
                    dispatch(setCurrentPath(location + '/'));
                    dispatch(fetchCurrentItem(location + '/'));
                })
                .catch((err) => {
                    dispatch({ type: RENAME_ITEM_FAILURE, payload: err });
                });
        });
    };
};

export const openCreateFileWindow = function() {
    return (dispatch) => {
        dispatch({ type: OPEN_CREATE_FILE_WINDOW });
    };
};

export const closeCreateFileWindow = function() {
    return (dispatch) => {
        dispatch({ type: CLOSE_CREATE_FILE_WINDOW });
    };
};

export const openCreateFolderWindow = function() {
    return (dispatch) => {
        dispatch({ type: OPEN_CREATE_FOLDER_WINDOW });
    };
};

export const closeCreateFolderWindow = function() {
    return (dispatch) => {
        dispatch({ type: CLOSE_CREATE_FOLDER_WINDOW });
    };
};

export const openConsentWindow = function() {
    return (dispatch) => {
        dispatch({ type: OPEN_CONSENT_WINDOW });
    };
};

export const closeConsentWindow = function() {
    return (dispatch) => {
        dispatch({ type: CLOSE_CONSENT_WINDOW });
    };
};

export const openRenameWindow = function(item) {
    return (dispatch) => {
        dispatch({ type: OPEN_RENAME_WINDOW, payload: item });
    };
};

export const closeRenameWindow = function() {
    return (dispatch) => {
        dispatch({ type: CLOSE_RENAME_WINDOW });
    };
};

export const createFolder = function(name, path) {
    return (dispatch) => {
        dispatch({ type: CREATE_FOLDER });
        const fileClient = new PodClient();
        return fileClient
            .create(path + name + '/')
            .then(() => {
                dispatch({ type: CREATE_FOLDER_SUCCESS });
                dispatch(setCurrentPath(path));
            })
            .catch((err) => {
                dispatch({ type: CREATE_FOLDER_FAILURE, payload: err });
            });
    };
};
