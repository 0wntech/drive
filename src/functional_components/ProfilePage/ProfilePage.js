import React from 'react';
import { connect } from 'react-redux';
import styles from './ProfilePage.module.css';
import { ClassicSpinner } from 'react-spinners-kit';

import Settings from '../../assets/svgIcons/Settings';
import EditIcon from '../../assets/svgIcons/Edit';
import defaultIcon from '../../assets/icons/defaultUserPic.png';
import KeyValuePair from '../KeyValuePair/KeyValuePair';

const ProfilePage = ({
    user
}) => {
    if (user) {
        return (
            <div className={styles.grid}>
                <div className={styles.toolbarArea}>
                    <div className={styles.iconWrapper}>
                        <Settings className={styles.settings} />
                    </div>
                </div>
                <div className={styles.profileContainer}>
                    <div className={styles.headContainer}>
                        <div
                            className={styles.profileImage}
                            style={
                                user && user.picture
                                    ? {
                                          backgroundImage: `url(${user.picture})`,
                                      }
                                    : { backgroundImage: `url(${defaultIcon})` }
                            }
                        />
                        <div className={styles.nameContainer}>
                            <div className={styles.nameLabel}>{user.name}</div>
                            <div className={styles.webIdLabel}>
                                {user.webId.replace('/profile/card#me', '')}
                            </div>
                        </div>
                        <div className={styles.bio}>
                            {user.bio ? user.bio : 'add Bio'}
                        </div>
                        <div className={styles.editWrapper}>
                            <EditIcon className={styles.editIcon} />
                        </div>
                    </div>
                    <KeyValuePair label="Job" value={user.job} />
                    <KeyValuePair label="Email" value={user.emails} />
                    <KeyValuePair label="Telephone" value={user.telephones} />
                </div>
            </div>
        );
    } else {
        return (
            <div className={styles.spinner}>
                <ClassicSpinner size={30} color="#686769" loading={!user} />
            </div>
        );
    }
};

const mapStateToProps = (state) => ({
    user: state.app.user,
});

export default connect(
    mapStateToProps,
    {}
)(ProfilePage);
