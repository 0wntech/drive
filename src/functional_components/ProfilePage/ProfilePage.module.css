.grid {
    height: 100%;
    display: grid;
    grid-template-rows: 40px 23fr;
    grid-template-columns: 3fr 6fr 3fr;
    grid-template-areas:
        'toolbar toolbar toolbar'
        '. profile .';
    background-color: #ffffff;
    border-radius: 20px;
}

.editButtons {
    display: flex;
    flex-direction: column;
}

.iconWrapper {
    height: 100%;
    display: flex;
    align-items: center;
    grid-area: settings;
    cursor: pointer;
}

.settings:hover path {
    fill: #c2c2c2;
}

.profileContainer {
    grid-area: profile;
    overflow-y: auto;
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100%;
    margin-top: 65px;
}

.profileImage {
    grid-area: profileImg;
    position: relative;
    width: 100%;
    height: 100%;
    background-size: cover;
    background-position: center;
    border-radius: 50%;
}

.nameContainer {
    grid-area: name;
    padding: 0 0 0 30px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}

.bioLabel {
    letter-spacing: 0.2em;
    font-family: Roboto;
    font-style: normal;
    font-weight: 300;
    font-size: 18px !important;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    line-height: 21px;
    width: 100% !important;
}

.bio {
    padding: 0 0 0 30px;
    grid-area: bio;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.headInfoContainer {
    width: 50%;
    margin-left: 20px;
}

.headContainer {
    width: 100%;
    max-height: 100%;
    margin-bottom: 20px;

    display: grid;
    grid-template-columns: 10vw 3fr 1fr;
    grid-template-rows: 5vw 5vw;
    grid-template-areas:
        'profileImg name editIcon'
        'profileImg bio .';
}

.iconContainer {
    width: 12%;
}

.toolbarArea {
    grid-area: toolbar;
    background-color: #e2e2e6;
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    width: 100%;
    display: grid;
    grid-template-columns: 11fr 1fr;
    grid-template-areas: '. settings';
}

.nameLabel {
    font-family: Roboto;
    font-style: normal;
    width: 100% !important;
    font-weight: 300;
    font-size: 24px !important;
    line-height: 28px;
    position: relative;
    letter-spacing: 0.2em;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: visible;
}

.webIdLabel {
    margin-top: 10px;
    font-family: Roboto;
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    line-height: 16px;
    letter-spacing: 0.2em;
    color: #454545;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}

.valueLabel {
    letter-spacing: 0.2em;
    font-family: Roboto;
    font-style: normal;
    font-weight: 300;
    font-size: 18px;
    line-height: 21px;
}

.editWrapper {
    grid-area: editIcon;
    display: flex;
    align-items: center;
    justify-content: center;
}

.icon {
    cursor: pointer;
}
