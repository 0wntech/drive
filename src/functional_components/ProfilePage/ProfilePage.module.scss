@import '../../styles/mixins.scss';
@import '../../styles/constants.scss';
.grid {
    height: 100%;
    display: grid;
    grid-template-columns: 2fr 6fr 2fr;
    grid-template-areas: '. profile .';
}

svg:hover {
    cursor: pointer;
}

.editButtons {
    display: flex;
    flex-direction: row;
}

.editWrapper {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 18px;
}

.iconWrapper {
    height: 100%;
    display: flex;
    align-items: center;
    cursor: pointer;
    z-index: 3;
}

.settings:hover path {
    fill: #c2c2c2;
}

.profileContainer {
    grid-area: profile;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    margin: 65px 0;
}

.profileImage {
    grid-area: profileImg;
    position: relative;
    width: 150px;
    height: 150px;
    background-size: cover;
    background-position: center;
    border-radius: 50%;
}

.editPhoto {
    grid-area: profileImg;
    position: absolute;
    width: 150px;
    height: 150px;
    display: flex;
    align-items: center;
    justify-content: center;
    background: rgba(5, 5, 5, 0.5);
    border-radius: 50%;
    z-index: 1;
}

.editPhoto:hover {
    cursor: pointer;
}

.editPhoto:hover {
    cursor: pointer;
}

.nameContainer {
    grid-area: name;
    padding: 0 0 0 30px;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.bioLabel {
    font-family: 'Helvetica Neue Light';
    font-style: normal;
    font-weight: 300;
    font-size: 18px;
    line-height: 21px;
    width: 100% !important;
    text-align: center;
}

.headDataContainer {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    flex: 1;
    width: 80%;
    align-items: center;
}

.headContainer {
    width: 100%;
    max-height: 100%;
    margin-bottom: 20px;
    display: flex;
    position: relative;
    flex-direction: column;
    align-items: center;
}

.iconContainer {
    width: 12%;
}

.toolbarArea {
    grid-area: toolbar;
    background-color: #e2e2e6;
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    width: 100%;
    display: grid;
    position: relative;
    grid-template-columns: 11fr 1fr;
    grid-template-areas: '. settings';
}

.toolbarHeader {
    letter-spacing: 0.05em;
    font-size: 18px;
    position: absolute;
    width: 100%;
    text-align: center;
    align-self: center;
}

.nameLabel {
    letter-spacing: 0.5px;
    font-family: 'Helvetica Neue Light';
    font-style: normal;
    width: 100% !important;
    font-weight: 300;
    font-size: 24px;
    line-height: 28px;
    text-align: center;
}

.webIdLabel {
    margin: 10px;
    font-family: 'Helvetica Neue Light';
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    color: #454545;
    text-align: center;
}

.valueLabel {
    font-family: 'Helvetica Neue Light';
    font-style: normal;
    font-weight: 300;
    font-size: 18px;
    line-height: 21px;
}

.icon {
    cursor: pointer;
}

.icon:hover path {
    fill: #c2c2c2;
}

.iconWhite {
    path {
        fill: #ffffff;
    }
    margin-right: 10px;
}

.multiline {
    @include multiLineEllipsis(
        $lineHeight: 1.2em,
        $lineCount: 2,
        $bgColor: white
    );
}

.actionButton {
    margin: 0 4px;
}

@media (max-width: $screen_m) {
    .grid {
        grid-template-areas: 'profile profile profile';
        grid-template-rows: 1fr;
    }

    .profileContainer {
        margin-top: 26px;
    }

    .nameLabel {
        margin-top: 10px;
        width: 100%;
        font-size: 16px;
        text-align: center;
        line-height: initial;
    }

    .webIdLabel {
        width: 100%;
        margin-top: 10px;
        font-size: 16px;
        margin-bottom: 3px;
    }

    .bioLabel {
        width: 100%;
        margin-top: 10px;
        font-size: 16px;
        text-align: center;
    }

    .multiline {
        padding-right: 0px;
        margin-right: 0px;
    }

    .profileImage {
        margin-right: 0px;
    }
}
