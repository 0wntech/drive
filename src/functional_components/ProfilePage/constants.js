export const responsiveContacts = {
    superLargeDesktop: {
        // the naming can be any, depends on you.
        breakpoint: { max: 4000, min: 3000 },
        items: 8,
    },
    desktop: {
        breakpoint: { max: 3000, min: 1800 },
        items: 6,
    },
    mediumDesktop: {
        breakpoint: { max: 1800, min: 1024 },
        items: 3,
    },
    smallDesktop: {
        breakpoint: { max: 1024, min: 768 },
        items: 2,
    },
    bigTablet: {
        breakpoint: { max: 768, min: 600 },
        items: 4,
    },
    tablet: {
        breakpoint: { max: 600, min: 464 },
        items: 3,
    },

    mobile: {
        breakpoint: { max: 464, min: 0 },
        items: 2,
    },
};

export const responsiveApps = {
    superLargeDesktop: {
        // the naming can be any, depends on you.
        breakpoint: { max: 4000, min: 3000 },
        items: 8,
    },
    desktop: {
        breakpoint: { max: 3000, min: 1800 },
        items: 6,
    },
    mediumDesktop: {
        breakpoint: { max: 1800, min: 1024 },
        items: 3,
    },
    smallDesktop: {
        breakpoint: { max: 1024, min: 768 },
        items: 2,
    },
    bigTablet: {
        breakpoint: { max: 768, min: 600 },
        items: 4,
    },
    tablet: {
        breakpoint: { max: 600, min: 464 },
        items: 3,
    },

    mobile: {
        breakpoint: { max: 464, min: 0 },
        items: 2,
    },
};
