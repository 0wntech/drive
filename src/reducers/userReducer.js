import {
    LOGIN,
    LOGIN_SUCCESS,
    LOGIN_FAIL,
    FETCH_USER,
    FETCH_USER_SUCCESS,
    FETCH_USER_FAIL,
    SET_WEBID,
    UPDATE_PROFILE,
    UPDATE_PROFILE_SUCCESS,
    UPDATE_PROFILE_FAILURE,
    CHANGE_PROFILE_PHOTO,
    CHANGE_PROFILE_PHOTO_SUCCESS,
    CHANGE_PROFILE_PHOTO_FAILURE,
    CLEAR_ERROR,
} from '../actions/types';

const INITIAL_STATE = {
    webId: null,
    user: null,
    loadLogin: false,
    loadUser: false,
    session: null,
    updatingProfile: false,
    updatingProfilePic: false,
    error: {
        LOGIN: false,
        FETCH_USER: false,
        UPDATE_PROFILE: false,
        CHANGE_PROFILE_PHOTO: false,
    },
};

export default (state = INITIAL_STATE, action) => {
    const { payload, type } = action;
    console.log('User Reducer got action: ', type, '\nValue: ', payload);
    switch (type) {
        case CLEAR_ERROR:
            return { ...state, error: INITIAL_STATE.error };
        case LOGIN:
            return {
                ...state,
                loadLogin: true,
                error: { ...state.error, LOGIN: false },
            };
        case LOGIN_SUCCESS:
            return {
                ...state,
                loadLogin: false,
                webId: payload.webId,
                session: payload,
                error: { ...state.error, LOGIN: false },
            };
        case LOGIN_FAIL:
            return {
                ...state,
                loadLogin: false,
                error: { ...state.error, LOGIN: payload },
            };
        case FETCH_USER:
            return {
                ...state,
                loadUser: true,
                error: { ...state.error, FETCH_USER: false },
            };
        case FETCH_USER_SUCCESS:
            return {
                ...state,
                loadUser: false,
                user: payload,
                error: { ...state.error, FETCH_USER: false },
            };
        case FETCH_USER_FAIL:
            return {
                ...state,
                loadUser: false,
                error: { ...state.error, FETCH_USER: payload },
            };
        case SET_WEBID:
            return { ...state, webId: payload };
        case UPDATE_PROFILE:
            return {
                ...state,
                updatingProfile: true,
                updateProfileError: false,
                error: { ...state.error, UPDATE_PROFILE: false },
            };
        case UPDATE_PROFILE_SUCCESS:
            return {
                ...state,
                updatingProfile: false,
                updateProfileError: false,
                error: { ...state.error, UPDATE_PROFILE: false },
            };
        case UPDATE_PROFILE_FAILURE:
            return {
                ...state,
                updatingProfile: false,
                error: { ...state.error, UPDATE_PROFILE: payload },
            };
        case CHANGE_PROFILE_PHOTO:
            return { ...state, updatingProfilePic: true };
        case CHANGE_PROFILE_PHOTO_SUCCESS:
            return { ...state, updatingProfilePic: false };
        case CHANGE_PROFILE_PHOTO_FAILURE:
            return { ...state, updatingProfilePic: false, error: payload };
        default:
            return state;
    }
};
