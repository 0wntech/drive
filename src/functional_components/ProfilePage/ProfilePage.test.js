import React from 'react';
import { ProfilePage } from './ProfilePage.js';
import { FETCH_USER_SUCCESS } from '../../actions/types';
import userReducer from '../../reducers/userReducer';
import TestRenderer from 'react-test-renderer';
describe('ProfilePage', () => {
    const user = {
        webId: 'https://bejow.owntech.de/profile/card#me',
        name: 'Ben',
        bio: 'This is bens Bio!',
        picture:
            'https://images-na.ssl-images-amazon.com/images/I/81-yKbVND-L._SY355_.png',
        emails: ['ben.test@owntech.de'],
        job: 'Software Engineer',
        telephones: ['0177313131'],
    };
    describe('ProfilePage Redux', () => {
        // passing undefined to get initial state
        const initialState = userReducer(undefined, {});
        it('receives user', () => {
            const result = userReducer(initialState, {
                type: FETCH_USER_SUCCESS,
                payload: user,
            });

            expect(result).toEqual({
                ...initialState,
                user: user,
            });
        });
    });
    describe('ProfilePage Snapshot', () => {
        it('matches the snapshot', () => {
            const tree = TestRenderer.create(<ProfilePage user={user} />);
            expect(tree.toJSON()).toMatchSnapshot();
        });
    });

    describe('ProfilePage UnitTest', () => {});
});
