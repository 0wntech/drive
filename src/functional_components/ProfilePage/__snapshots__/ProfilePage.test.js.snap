// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage ProfilePage Snapshot matches the snapshot 1`] = `
<Layout
  className="grid"
  hideToolbar={false}
  isLoading={false}
  label="Profile"
>
  <div
    className="profileContainer"
  >
    <div
      className="headContainer"
    >
      <div
        className="profileImage"
        style={
          Object {
            "backgroundImage": "url('https://images-na.ssl-images-amazon.com/images/I/81-yKbVND-L._SY355_.png')",
          }
        }
      />
      <div
        className="headDataContainer"
      >
        <div>
          <SingleValue
            className="nameLabel"
            dataKey="name"
            editable={false}
            placholder="Enter Name.."
            setValue={[Function]}
            value="Ben"
          />
          <div
            className="webIdLabel"
          >
            https://bejow.owntech.de
          </div>
        </div>
        <SingleValue
          className="bioLabel"
          editable={false}
          maxInput={256}
          placeholder="Add bio.."
          setValue={[Function]}
          value="This is bens Bio!"
        />
      </div>
      <div
        className="editWrapper"
      >
        <IconButton
          color="blue"
          onClick={[Function]}
          size="lg"
        >
          <SvgEdit
            className="iconWhite"
            data-test-id="edit"
            height="20"
            onClick={[Function]}
            viewBox="3 2 30 30"
            width="20"
          />
          Edit Profile
        </IconButton>
      </div>
    </div>
    <KeyValuePair
      dataKey="job"
      editable={false}
      label="Job:"
      placeholder="Software Engineer"
      setValue={[Function]}
      value="Software Engineer"
    />
    <KeyValuePair
      dataKey="emails"
      editable={false}
      label="Email:"
      placeholder={
        Array [
          "ben.test@owntech.de",
        ]
      }
      setValue={[Function]}
      value="ben.test@owntech.de"
    />
    <KeyValuePair
      dataKey="telephones"
      editable={false}
      label="Phone:"
      placeholder={
        Array [
          "0177313131",
        ]
      }
      setValue={[Function]}
      value="0177313131"
    />
  </div>
</Layout>
`;
